src/mjolnir/ingest_transit.cc

```diff
@@ -162,66 +163,57 @@ std::priority_queue<tileTransitInfo> select_transit_tiles(const boost::property_
       LOG_INFO("Done Reading Feed");
       const auto& stops = feed.get_stops();
 
+      // 1st pass to add all the stations, so we can add stops to its children in a 2nd pass
       for (const auto& stop : stops) {
-        if (stop.stop_id.empty()) {
-          continue;
+        if (stop.location_type == gtfs::StopLocationType::Station) {
+          auto& tile_info = get_tile_info(stop);
+          tile_info.stations.insert({stop.stop_id, feed_path});
         }
-        gtfs::Id currStopId = stop.stop_id;
-        float x = stop.stop_lon;
-        float y = stop.stop_lat;
-
-        GraphId currId =
-            GraphId(tile_level.tiles.TileId(y, x), TileHierarchy::GetTransitLevel().level, 0);
-
-        tileTransitInfo currTile;
+      }
```

Comment:
this is the real change: we do a first pass over the GTFS stops to just add all stations so we can determine the proper children for them in the second pass below.

tbh, it could still be simplified quite a bit IMO, but I'll look at it again some other time.

Reply:
by doing it this way you have, im pretty sure, created a bug. if a trip begins at at station (ie. `gtfs::StopLocationType::Station`) you will not record it below because you have pulled the loops apart. one options would be, when we load the feed we do so with a derived class and do a quick sort on the stops by their type. something like this:

```c++
class sortable_feed : public just_gtfs::feed {
public:
 void sort_stops() {
   // subtract to make station the smallest in the enum
   std::sort(stops.begin(), stops.end(), [](const auto& a, const auto& b){ return static_cast<uint8_t>(a.location_type - 1) < static_cast<uint8_t>(b.location_type - 1); }
 }
};
...
// above near the feed reading part just call sort once
feed.sort_stops();
...
// then everything can stay in the same for loop and we dont have to check the types and all that
```

or maybe you can think of a cleaner way to loop over the stations first? i guess you could make an outer loop with the types you wnat to look for in each iteration, that might be clean as well? up to you, but i think there is a new bug here if we dont handle trips from stations :smile:

Reply:
I'm not really sure it's a bug.. Sounds more like GTFS is designed in a way that a station can't be referenced by trips by a `stop_id`: https://developers.google.com/transit/gtfs/reference#stop_timestxt.

> `stop_id`: [...] Referenced locations must be stops, not stations or station entrances. [...]

Though I'm not saying it's not going to be the case for some shoddy feed, but the spec sounds pretty clear or am I missing smth? Do you still want to protect for that?

Reply:
it's possible I am misremebering. our previous interface to gtfs was transitland and I thought that the station always existed and the platform and egress were faked when not there. seems like the spec says platform is always there but station and egress may not be?

Reply:
I think it's more trying to force a feed to be reasonable? I was actually surprised that a station should directly be referenced in a trip, it didn't seem intuitive to me (same for egresses no?). From what I understood, a station might not be there (e.g. a simple bus stop), but every station that is there, needs to have at least one platform. Egresses are probably optional as well?

I don't know why transit.land would fake a platform, you're right, and we're still doing that. Think I'll add a log to that so we can see it in real GTFS feeds.

Reply:
seems to me we s hould look at some feeds with bus schedules in them to see how they are modeled for real.

but at least gtfs spec says regarding `parent_id`
> • Optional for stops/platforms (location_type=0).

which i guess means our understanding is correct regardless of what transitlands model had. but does definitely mean we need to handle `parent_id` being empty string from just_gtfs when its a stop/platform

Reply:
> but does definitely mean we need to handle parent_id being empty string from just_gtfs when its a stop/platform

We already do handle that, see line 180 onwards in this PR.